lib/index.js

```diff
@@ -469,6 +471,55 @@ class AuthenticationPlugin {
     return this.getUsersRepository().delete(user._id, {refresh: 'wait_for'});
   }
 
+  /**
+   * Search users by ES query concerning exclusively their username
+   *
+   * @param {Object} searchBody - Search request body (query, sort, etc.)
+   * @param {Object} options - from (undefined), size (undefined)
+   *
+   * @returns {Promise.<{ hits, total }>}
+   */
+  async search (searchBody, { from, size } = {}) {
```

Comment:
(question) In Kuzzle's API, search query bodies can usually be left empty. What happens if that's the case here: do we have an object, or a `null` value? If it's the latter, then this method will throw when running `objectRecursiveMap`

Reply:
It's currently a "match_all" if any request doesn't contain 'username' field (included if left empty), I will change that into only if left empty and ForbiddenError otherwise

Reply:
That's not what I meant. What happens if no search body is provided? Does the plugin receive an empty object, or `null`?

If your method receives a `null` values, it will throw.

Reply:
Oh I get it, it will always receive an empty object thanks to `getSearchParams()` which use [getSearchBody](https://github.com/kuzzleio/kuzzle/blob/master/lib/api/request/kuzzleRequest.ts#L698)